picturesafe-search-spring-boot-demo/src/main/java/de/picturesafe/search/starterdemo/StarterDemoApplication.java

```diff
@@ -36,9 +40,25 @@ public static void main(String[] args) {
         SpringApplication.run(StarterDemoApplication.class, args);
     }
 
-    @Override
-    public void run(String... args) throws Exception {
-        final String elasticsearchServerVersion = singleIndexElasticsearchService.getElasticsearchInfo().getServerVersion();
-        LOGGER.info("Elasticsearch version = {}", elasticsearchServerVersion);
+    @Bean
+    public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
+        return args -> {
+            logBeanDefinitions(ctx);
+
+            final String elasticsearchServerVersion = singleIndexElasticsearchService.getElasticsearchInfo().getServerVersion();
+            LOGGER.info("Elasticsearch version = {}", elasticsearchServerVersion);
+        };
+    }
+
+    private void logBeanDefinitions(ApplicationContext ctx) {
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Running Spring Boot app with following bean definitions:");
+            final String[] beanNames = ctx.getBeanDefinitionNames();
+            Arrays.sort(beanNames);
+            for (String beanName : beanNames) {
```

Comment:
Checkstyle: missing final

Reply:
I did not see any checkstyle violation...